crates/subspace-farmer/src/dsn/tests.rs

```diff
@@ -173,20 +173,23 @@ async fn test_dsn_sync() {
             listen_on: vec!["/ip4/127.0.0.1/tcp/0".parse().unwrap()],
             dsn_sync: false,
         },
-        seeder_max_plot_size * PIECE_SIZE as u64,
+        // XXX: farmer reserves 8% for its own needs, so we need to update piece count here
+        seeder_max_plot_size * 100 / 92 * PIECE_SIZE as u64,
         seeder_max_plot_size,
         plot_factory,
         false,
     )
     .await
     .unwrap();
 
-    {
+    let piece_index_hashes = {
+        let pieces_per_segment = 262;
```

Comment:
One segment always has 256 pieces, where does 262 number come from?

Reply:
I just debugged everything, and it somehow is 262 for me (maybe because of the metadata settings). Should I fix it?

Reply:
Then something isn't matching chain parameters in this test, it is always 256 in the archiver on the node (which you're not using, so you didn't catch it there in other places).